pkg/cmd/cluster.go

```diff
@@ -84,35 +93,123 @@ func doStartCluster(cmd *cobra.Command, args []string) error {
 			return err
 		}
 
-		Dialog.Info("Initializing relay-core; this might take a couple minutes...")
 		initOpts := dev.InitializeOptions{
-			ImageRegistryPort: registryPort,
+			ImageRegistryPort: opts.ImageRegistryPort,
 		}
-		if err := dm.InitializeRelayCore(ctx, initOpts); err != nil {
+
+		if err := dm.Initialize(ctx, initOpts); err != nil {
 			return err
 		}
 
-		Dialog.Infof("Cluster connection can be used with: !Connection {type: kubernetes, name: %s}", dev.RelayClusterConnectionName)
+		Dialog.Info("Relay dev cluster is ready to use")
 	} else {
-		if err := cm.Start(ctx); err != nil {
-			return err
-		}
+		Dialog.Info("Cluster already exists")
+	}
 
-		cl, err := cm.GetClient(ctx, cluster.ClientOptions{Scheme: dev.DefaultScheme})
-		if err != nil {
-			return err
-		}
+	return nil
+}
 
-		// dev manager depends on a kubernetes client, which we can't get if a
-		// cluster doesn't exist, so we can't create one at the top of this
-		// function.
-		dm := dev.NewManager(cm, cl, DevConfig)
+func newInitClusterCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "initialize",
+		Aliases: []string{"init"},
+		Short:   "Initialize the local cluster",
+		RunE:    doInitCluster,
+	}
 
-		if err := dm.StartRelayCore(ctx); err != nil {
-			return err
+	return cmd
+}
+
+func doInitCluster(cmd *cobra.Command, args []string) error {
+	ctx := cmd.Context()
+
+	opts := cluster.InitializeOptions{}
+
+	return initCluster(ctx, opts)
+}
+
+func initCluster(ctx context.Context, opts cluster.InitializeOptions) error {
+	cm := cluster.NewManager(ClusterConfig)
+
+	if exists, _ := cm.Exists(ctx); !exists {
+		Dialog.Info("Cluster does not exist")
```

Comment:
Perhaps this should imply `create` if the referenced cluster doesn't exist, maybe with default configuration?

Reply:
I'll write up a specific issue covering this and the above comment, regarding improving the user flow of the dev commands. I'm conflicted on both points. I want to avoid duplicating the command args in both places (and perhaps as suggested we just use defaults if `init` is called first. We also need to determine how to ensure that the `init` phase is either rerunnable or blocked appropriately if already initialized.